hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -369,12 +372,13 @@ public DatanodeDeletedBlockTransactions getTransactions(
               if (checkInadequateReplica(replicas, txn)) {
                 continue;
               }
+              metrics.setNumBlockDeletionTransactionDataNodes(dnList.size());
```

Comment:
DnList **do not** have DNs which are unhealthy Or overloaded causing pending request as present. Normally these are not case.
`NumBlockDeletionTransactionDataNodes` metrics may not be useful at this point. This either can be added after DN filter is done at SCMBlockDeletingService.call() Or may not be required, since here, its called with same value in loop.

Reply:
From the Metrics given in this PR https://github.com/apache/ozone/pull/7249#issue-2554897610, the overloaded DNs were not a rare case, because before applying this patch, they EC key deletion was very slow, which should be cause by the large number of overloaded DNs. So I think we can keep this Metric.
By default this Metric be updated per minute, as long as the acquisition Metric rate is less than this interval, then we can monitor the overloaded DNs count correctly.

Reply:
@xichen01 Since dnList.size() is constant for the iteration, setting this inside loop of getTransaction() may not be redundant. We can move out to place where final dnList is prepared.

Reply:
I moved `metrics.setNumBlockDeletionTransactionDataNodes(dnList.size());` to after DeletedBlockLogImpl#getTransaction.